### PR TITLE
gh-104683: Argument Clinic: Params now render their own docstrings

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-104683

### CODE REVIEW

File: Lib/test/test_clinic.py [hunk 1]
```diff
@@ -834,8 +834,8 @@ def expect_failure(self, block, err, *, filename=None, lineno=None):
 
     def checkDocstring(self, fn, expected):
         self.assertTrue(hasattr(fn, "docstring"))
-        self.assertEqual(fn.docstring.strip(),
-                         dedent(expected).strip())
+        self.assertEqual(dedent(expected).strip(),
+                         fn.docstring.strip())
```
Comment (erlend-aasland) [inline Lib/test/test_clinic.py:838]: Strictly not needed, but really helps when docstring tests fail.

File: Tools/clinic/clinic.py [hunk 2]
```diff
@@ -5681,22 +5690,10 @@ def add_parameter(text: str) -> None:
...
-            add(param.name)
-            add('\n')
-            stripped = rstrip_lines(docstring)
-            add(textwrap.indent(stripped, "    "))
-        if text:
+        add, output = text_accumulator()
+        docstrings = [p.render_docstring() for p in params if p.docstring]
+        for docstring in docstrings:
+            add(docstring)
             add('\n')
```
Comment (AlexWaygood) [inline Tools/clinic/clinic.py]:
```suggestion
        for p in params:
            if p.docstring:
                add(p.render_docstring())
                add('\n')
```

File: Tools/clinic/clinic.py [hunk 3]
```diff
@@ -2775,6 +2775,15 @@ def get_displayname(self, i: int) -> str:
         else:
             return f'"argument {i}"'
 
+    def render_docstring(self) -> str:
+        if not self.docstring:
+            return ""
```
Comment (AlexWaygood) [inline Tools/clinic/clinic.py]: Oh, is this early return actually necessary? It looks like we only call `render_docstring()` if `self.docstring` is truthy. Maybe we could make it an assertion?

```suggestion
        assert self.docstring, "Shouldn't call this method unless self.docstring is truthy"
```

**Review by AlexWaygood — APPROVED** [review]
Looks good (left one more optional suggestion!)

File: Tools/clinic/clinic.py (hunk 3 above)
Comment (erlend-aasland) [reply Tools/clinic/clinic.py]: Indeed, this check is now obsolete. I believe we can just remove it.

**Comment by erlend-aasland** [comment]
Thanks for the review!

**Pull request merged**